tortoise/tortoise.go

```diff
@@ -653,7 +649,7 @@ func (t *turtle) onBallot(ballot *types.Ballot) error {
 		t.logger.With().Warning("base ballot not in state",
 			log.Stringer("base", ballot.Votes.Base),
 		)
-		return nil
+		return nil, nil
```

Comment:
regarding all the above `return nil, nil` lines.
so if a ballot's base ballot is outside the window, the node will not store it in db (in mesh.go)? 
shouldn't we try harder to look up db instead?

Reply:
it doesn't make a difference to find a base ballot without decoded votes, this is why there is sliding window

Reply:
protocol should define max distance for base ballot, sliding window should be alteast as large as that distance. i guess that should be a part of the config validation

Reply:
in that case, should the window-size be part of genesis config?

Reply:
max base distance should be in the genesis config or just a constant protocol parameter, and sliding window should be atleast as large as max base distance